### ISSUE TITLE
Models do not keep their deleted status in some cases

### ISSUE THREAD

**Issue by kltm** [geneontology/noctua#485]
_From @BarbaraCzub on August 30, 2017 11:40_

How do I delete this model?

I previously changed its state to 'delete' and marked it as 'Deprecated', but when I logged into Noctua this morning the state for this model was displayed as 'development' again.

I will not be doing any more work on this model, hence I marked it as 'delete', and made a new one instead (gomodel:598826eb00000503).

 (optional back ref: http://noctua.berkeleybop.org/editor/graph/gomodel:598826eb00000261)

cc: @RLovering 

_Copied from original issue: geneontology/noctua-models#42_

**Comment by kltm** [comment]
Tagging @BarbaraCzub back on.

**Comment by kltm** [comment]
That is all you need to do. At some future date a purging process will be added, as well as a default of hiding them, but in general curators can only request the delete.
In some cases, we may choose to manually destroy a model immediately, but that will be rare.

**Comment by BarbaraCzub** [comment]
If this is all that I needed to do, then this model should have been deleted. But instead its state had been changed to 'development' again (and not by me) before I changed it back to 'delete' yesterday.

**Comment by kltm** [comment]
I would suspect that you did not save the model last time, so when the index was reset recently it reverted. Also, looking at this, if it's possible that this got in through the SynGO import that we do, it could have been reset then as well. It's also possible that there is an unknown-until-now- issue in minerva. We'll be able to eliminated that one tonight.

**Comment by kltm** [comment]
Coming from https://github.com/geneontology/noctua-models/issues/42#issuecomment-326228230

I currently suspect that these issues are to do with you not saving your activity in models. If a model is created and not saved, it will eventually "disappear"; if a models is saved, has its state set to `delete` and it not saved again, that will eventually "revert" the model to the un-deleted state. Any action on a model that is not saved will eventually be reverted to the last saved state.

Looking at http://noctua.berkeleybop.org/editor/graph/gomodel:598826eb00000261 right now, the model state is `delete`, but the model itself has been unsaved. Looking at GitHub (https://github.com/geneontology/noctua-models/blob/master/models/598826eb00000261.ttl), the last time the model was saved was two days ago, and it is in `development` (https://github.com/geneontology/noctua-models/blob/master/models/598826eb00000261.ttl#L49).

**Comment by cmungall** [comment]
@kltm @vanaukenk @ukemi : the docs need to be clear about the need to save. There is no autosave. We probably need a FAQ for noctua as well that clearly states this.

**Comment by vanaukenk** [comment]
Okay.  
It seems the specific issue here is that ANY action on a model needs to be saved, including changing the status.

**Comment by kltm** [comment]
We have:
"Understanding "saving" on noctua.berkeleybop.org
"Naming your Model and Saving your Work" in the http://noctua.berkeleybop.org/doc/quick-start-guide

**Comment by BarbaraCzub** [comment]
Ok, I have changed this model back to the 'delete' state now, and I have definitely saved. Thanks.

**Comment by suzialeksander** [comment]
Adding to this ticket as 5fadbcf000000262 keeps popping back as production even though we're trying to delete it, so a very similar issue. 
As part of annotation review, we deleted an annotation which resulted in the model being blank.  But, noctua doesn't allow saving of blank models, so are we forced to add back anything just to mark a model as delete?
 
 
potential solution: noctua doesn't allow blank models to be saved UNLESS the state is delete?
